airsonic-main/src/main/java/org/airsonic/player/util/Util.java

```diff
@@ -133,6 +136,24 @@ public static String getURLForRequest(HttpServletRequest request) {
         return url;
     }
 
+    /**
+     * Return an URL for the given HTTP request, with anonymized sensitive parameters.
+     *
+     * @param request An HTTP request instance
+     * @return The associated anonymized URL
+     */
+    public static String getAnonymizedURLForRequest(HttpServletRequest request) {
+
+        String url = getURLForRequest(request);
+        UriComponentsBuilder builder = UriComponentsBuilder.fromUriString(url);
+        MultiValueMap<String, String> components = builder.build().getQueryParams();
+
+        // See RESTRequestParameterProcessingFilter
+        if (components.containsKey("p")) builder.replaceQueryParam("p", URL_SENSITIVE_REPLACEMENT_STRING);
```

Comment:
Wouldn't it would make sense to anonymize the username too?

Reply:
Yup, you're correct! Done!